app/src/main/kotlin/com/softteco/template/ui/feature/bluetooth/BluetoothScreen.kt

```diff
@@ -46,35 +45,39 @@ fun BluetoothScreen(
     val state by viewModel.state.collectAsState()
     val deviceName = stringResource(R.string.device_name)
     viewModel.filteredName = deviceName
-    BluetoothHelper.onScanResult = {
+
+    viewModel.provideOnScanCallback {
         viewModel.addScanResult(it)
     }
-    BluetoothHelper.onConnect = {
+
+    viewModel.provideOnConnectCallback {
         scope.launch {
```

Comment:
The coroutine is redundant here

Reply:
Required for redirection in the main thread.